controllers/rack.go

```diff
@@ -752,22 +768,25 @@ func (r *SingleClusterReconciler) scaleDownRack(
 
 	pod = utils.GetPod(podName, oldPodList.Items)
 
-	// Ignore safe stop check on pod not in running state.
-	if utils.IsPodRunningAndReady(pod) {
+	isPodRunningAndReady := utils.IsPodRunningAndReady(pod)
+
+	// Ignore safe stop check if pod is not running.
+	// Ignore migrate-fill-delay if pod is not running. Deleting this pod will not lead to any migration.
+	if isPodRunningAndReady {
 		if res := r.waitForMultipleNodesSafeStopReady([]*corev1.Pod{pod}, ignorablePods, true); !res.isSuccess {
 			// The pod is running and is unsafe to terminate.
 			return found, res
 		}
-	}
 
-	// set migrate-fill-delay to 0 across all nodes of cluster to scale down fast
-	// temporarily add the pod to be deleted in ignorablePods slice to avoid setting migrate-fill-delay to pod when
-	// pod is not ready
-	if res := r.setMigrateFillDelay(
-		policy, &rackState.Rack.AerospikeConfig, true,
-		append(ignorablePods, *pod),
-	); !res.isSuccess {
-		return found, res
+		// set migrate-fill-delay to 0 across all nodes of cluster to scale down fast
+		// temporarily add the pod to be deleted in ignorablePods slice to avoid setting migrate-fill-delay to pod when
+		// pod is not ready
```

Comment:
Need to update the comment here as we are not setting MFD only in-case of running pod
```suggestion
		// temporarily add the pod to be deleted in ignorablePods slice to avoid setting migrate-fill-delay to pod being deleted
```